appengine/storage.js

```diff
@@ -71,6 +71,15 @@ BlocklyStorage.restoreBlocks = function(opt_workspace) {
 BlocklyStorage.link = function(opt_workspace) {
   var workspace = opt_workspace || Blockly.getMainWorkspace();
   var xml = Blockly.Xml.workspaceToDom(workspace, true);
+  // Remove x/y coordinates from XML if there's only one block stack.
+  // There's no reason to store this, removing it helps with anonymity.
+  if (workspace.getTopBlocks(false).length == 1 && xml.querySelector) {
+    var block = xml.querySelector('block');
+    if (block) {
```

Comment:
The existence of a block would appear to be guaranteed (if workspace has one top block), but technically it might be a shadow.  Thus it's better to check than to blow up.  I don't consider it worth adding code to strip coordinates from a single top-level shadow block.